CacheService/docker_context/CacheHandler.py

```diff
@@ -0,0 +1,205 @@
+            return goodResult(msg="Stock Data", data=data['stock_amount']), 200
+        else:
+            result, status = await self.RedisHandler.updateStockCache(user_id, stock_id)
+            if status == 200:
+                data = await self.RedisHandler.rGet(f'{user_id}_{stock_id}')
+                return goodResult(msg="Stock Data", data=data['stock_amount']), 200
+            else:
+                return result, status
+
+    async def addFunds(self, trans_num, command, user_id, funds):
+        # TODO Update DBM to take trans_num and commands
+        data = {
+            "user_id": user_id,
+            "funds": float(funds)
+        }
+        result, status = await self.client.postRequest(f"{self.dbmURL}/funds/add_funds", data)
+        if status == 200:
+            logger.debug(f'{__name__} - Successfully added funds, caching')
+            result, status = await self.RedisHandler.updateAccountCache(user_id)
+        return result, status
+
+    async def buyStocks(self, trans_num, command, user_id, stock_symbol, stock_amount, total_value):
+        data = {
+            "transaction_num": trans_num,
+            "command": command,
+            "user_id": user_id,
+            "stock_id": stock_symbol,
+            "amount": total_value,
+            "amount_of_stock": stock_amount,
+            "time": currentTime()
+        }
+        await self.RedisHandler.rSet(f"{user_id}_BUY", data)
+        return goodResult(msg="Buy created", data=''), 200
+
+    async def sellStocks(self, trans_num, command, user_id, stock_symbol, stock_amount, total_value):
+        data = {
+            "transaction_num": trans_num,
+            "command": command,
+            "user_id": user_id,
+            "stock_id": stock_symbol,
+            "amount": total_value,
+            "amount_of_stock": stock_amount,
+            "time": currentTime()
+        }
+        await self.RedisHandler.rSet(f"{user_id}_SELL", data)
+        return goodResult(msg="Sell created", data=''), 200
+
+    async def getBuyStocks(self, user_id):
+        check = await self.RedisHandler.rExists(f"{user_id}_BUY")
+        if check:
+            return goodResult(msg="Return buy stock", data=await self.RedisHandler.rGet(f"{user_id}_BUY")), 200
+        else:
+            logger.info(f'{__name__} - Cannot get stocks for buy for {user_id} as it does not exist in cache')
+            return errorResult(err="Buy doesn't exist", data=''), 404
+
+    async def getSellStocks(self, user_id):
+        check = await self.RedisHandler.rExists(user_id + "_SELL")
+        if check:
+            return goodResult(msg="Return sell stock", data=await self.RedisHandler.rGet(f"{user_id}_SELL")), 200
+        else:
+            logger.info(f'{__name__} - Cannot get stocks for sell for {user_id} as it does not exist in cache')
+            return errorResult(err="Sell doesn't exist", data=''), 404
+
+    async def commitBuyStocks(self, trans_num, command, user_id):
```

Comment:
That logic resides in the Transaction Service since it is more specific to the transaction itself

Reply:
Mitchell is correct here, that's occuring in the transaction service. I could change it so that it happens in the cache though (probably not a terrible idea to cut down API calls).

Reply:
But I believe we wanted to cut down on the actual checks that the cache was doing, which is why we put it in the transaction server. It would be quick and easy to swap over though

Reply:
Ya I am thinking as an optimization we move these checks to a cache right by the web service. We would do something similar for the triggers as well.